lib/rules/jsx-no-invalid-rel.js

```diff
@@ -0,0 +1,190 @@
+/**
+ * @fileoverview Forbid rel attribute to have non-valid value
+ * @author Sebastian Malton
+ */
+
+'use strict';
+
+const docsUrl = require('../util/docsUrl');
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+const standardValues = new Map(Object.entries({
+  alternate: new Set(['link', 'area', 'a']),
+  author: new Set(['link', 'area', 'a']),
+  bookmark: new Set(['area', 'a']),
+  canonical: new Set(['link']),
+  'dns-prefetch': new Set(['link']),
+  external: new Set(['area', 'a', 'form']),
+  help: new Set(['link', 'area', 'a', 'form']),
+  icon: new Set(['link']),
+  license: new Set(['link', 'area', 'a', 'form']),
+  manifest: new Set(['link']),
+  modulepreload: new Set(['link']),
+  next: new Set(['link', 'area', 'a', 'form']),
+  nofollow: new Set(['area', 'a', 'form']),
+  noopener: new Set(['area', 'a', 'form']),
+  noreferrer: new Set(['area', 'a', 'form']),
+  opener: new Set(['area', 'a', 'form']),
+  pingback: new Set(['link']),
+  preconnect: new Set(['link']),
+  prefetch: new Set(['link']),
+  preload: new Set(['link']),
+  prerender: new Set(['link']),
+  prev: new Set(['link', 'area', 'a', 'form']),
+  search: new Set(['link', 'area', 'a', 'form']),
+  stylesheet: new Set(['link']),
+  tag: new Set(['area', 'a'])
+}));
+const components = new Set(['link', 'a', 'area', 'form']);
+
+function splitIntoRangedParts(node) {
+  const res = [];
+  const regex = /\s*([^\s]+)/g;
+  const valueRangeStart = node.range[0] + 1; // the plus one is for the initial quote
+  let match;
+
+  // eslint-disable-next-line no-cond-assign
+  while ((match = regex.exec(node.value)) !== null) {
+    const start = match.index + valueRangeStart;
+    const end = start + match[0].length;
+    res.push({
+      reportingValue: `"${match[1]}"`,
+      value: match[1],
+      range: [start, end]
+    });
+  }
+
+  return res;
+}
+
+function checkLiteralValueNode(context, node, parentNodeName) {
+  if (typeof node.value !== 'string') {
+    return context.report({
+      node,
+      messageId: 'onlyStrings',
```

Comment:
tests are failing because eslint@4 does not support messageId and template messages.